src/main/resources/com/google/api/codegen/go/discovery_fragment.snip

```diff
@@ -9,101 +9,136 @@
+       serviceTitle = context.getService.getTitle
+    package main
+    
+    // BEFORE RUNNING:
+    // ---------------
+    // 1. If not already done, enable the {@serviceTitle}
+    //    and check the quota for your project at
+    //    https://console.developers.google.com/apis/api/{@apiName}
+    // 2. This sample uses Application Default Credentials for authentication.
+    //    If not already done, install the gcloud CLI from
+    //    https://cloud.google.com/sdk/ and run
+    //    `gcloud beta auth application-default login`
+    // 3. Install and update the Go dependencies by running `go get -u` in the
+    //    project directory.
+    
+    import (
+      "encoding/json"
+      "fmt"
+      "log"
+    
+      "golang.org/x/net/context"
+      "golang.org/x/oauth2/google"
+    
+      "google.golang.org/api/{@apiName}/{@apiVersion}"
+    )
+    
+    func main() {
+      ctx := context.Background()
+      # TODO(pongad): Every API except autoscaler has the string as API.CloudPlatformScope, find out why
+
+      // Authentication is provided by the `gcloud` tool when running locally
+      // and by built-in service accounts when running on GAE, GCE, or GKE.
+      // For more information, see
+      // https://developers.google.com/identity/protocols/application-default-credentials
+      {@clientName()}, err := google.DefaultClient(ctx{@scopeArg(method)})
+      if err != nil {
+        log.Fatal(err)
+      }
+
+      // Construct the {@apiName} service object for interacting with the API.
+      // You can browse other available API services and versions at
+      // https://github.com/google/google-api-go-client
+      {@serviceName()}, err := {@apiName}.New({@clientName()})
+      if err != nil {
+        log.Fatal(err)
+      }
+
+      {@initialize(method)}
+      @if context.isPageStreaming(method)
+        {@iterableCall(method)}
+      @else
+        {@compactCall(method)}
+      @end
+    }
   @end
-}
 @end
 
 @private scopeArg(method)
   @let scopeList = context.getAuthScopes(method)
     @if not(scopeList.isEmpty)
-    , {@context.getApi.getName}.{@scopeList.get(0)}Scope
+      , {@context.getApi.getName}.{@scopeList.get(0)}Scope
     @end
   @end
 @end
 
 @private compactCall(method)
   @if context.isResponseEmpty(method)
-    @if context.hasRequestField(method)
-      call := {@methodPath(method)}({@callArg(method)}).Context(ctx)
-      if err := call.Do(); err != nil {
-        // {@TODO()} Handle error.
-      }
-    @else
-      if err := {@methodPath(method)}({@callArg(method)}).Context(ctx).Do(); err != nil {
-        // {@TODO()} Handle error.
-      }
-    @end
+    if err := {@request(method)}.Context(ctx).Do(); err != nil {
+      log.Fatal(err)
+    }
   @else
-    # TODO(pongad): generate a better name for "resp"
-    resp, err := {@methodPath(method)}({@callArg(method)}).Context(ctx).Do()
+    {@responseName()}, err := {@request(method)}.Context(ctx).Do()
+    if err != nil {
+      log.Fatal(err)
```

Comment:
After discussion with @saicheems , we both felt that `log.Fatal` was more usable for the most critical use case, that of trying to get an API call working for the first time. In the previous formulation, these error handlers generally accounted for a majority of TODOs in the Go samples. And in cases where they actually result in an error, the user would be required to develop them before seeing error messages. The current behavior seems more useful, providing a similar experience to the samples in the other languages (with exception-like error handling).

If others strongly disagree, I propose a combination as follows, which will preserve the desirable features while still alerting the user with additional TODOs:
```
if err != nil {
  // TODO: Change code below to handle error.
  log.Fatal(err)
}
```